### PR TITLE
Variadic option arguments for Test and Step functions

### PR DESCRIPTION
**Pull request by Ubermensch01**
This is a working version for the variadic approach to handling options for Test and Step functions that allows us to increase functionality without breaking existing codebase and with less impact to the existing test suites.

### CODE REVIEW

File: example/example_functions.go [hunk 1]
```diff
@@ -1,17 +1,26 @@
...
-	})
+	}))
 }
 
 func doSomethingNested() {
-	allure.Step("Because we can!", func() {})
+	allure.Step(allure.Description("Because we can!"), allure.Body(func() {}))
 }
 
 func addSomeParameters(parameters map[string]interface{}) {
```
Comment (MaximeBellou) [inline example/example_functions.go]: It looks like this is not used. What is the purpose of this method?

File: options.go [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+	}
+}
+
+func Feature(feature string) Option {
+	return func(r hasOptions) {
+		r.addLabel("feature", feature)
+	}
+}
+
+func Parameter(name string, value interface{}) Option {
```
Comment (MaximeBellou) [inline options.go]: Can you add a func called Parameters that take as argument a map[string]interface{}, and iterate to call addParameter for each interface?
I personally prefer this approach when we have to pass multiple parameters, so having the 2 possibilities would be great

File: options.go [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+	}
+}
+
+func Description(description string) Option {
+	return func(r hasOptions) {
+		r.addDescription(description)
+	}
+}
+
+func Body(action func()) Option {
```
Comment (MaximeBellou) [inline options.go]: Could we rename it into Action? Body is more associated to API requests content, actions is more meaningful for test scenarios.

File: example/example_functions.go (hunk 1 above)
Comment (Ubermensch01) [reply example/example_functions.go]: This was meant to illustrate a programmatic way of specifying parameters, but I removed a call to this function from the example test scripts.

File: options.go (hunk 3 above)
Comment (Ubermensch01) [reply options.go]: Renamed. I wasn't completely sure with `allure.Body` either.

File: options.go (hunk 2 above)
Comment (Ubermensch01) [reply options.go]: Implemented.

**Pull request merged**